lib/mogrify.ex

```diff
@@ -275,11 +276,22 @@ defmodule Mogrify do
     image_operator(image, "xc:#{color}")
   end
 
+  def add_option(image, option) do
+    validate_option(option)
+    custom(image, option.name, option.argument)
+  end
+
   def custom(image, action, options \\ nil) do
     %{image | operations: image.operations ++ [{action, options}]}
   end
 
   def image_operator(image, operator) do
     %{image | operations: image.operations ++ [{:image_operator, operator}]}
   end
+
+  defp validate_option(%Option{name: name, require_arg: true, argument: nil}) do
+    option_name = name |> String.replace("-", "")
```

Comment:
Options may start with a `+` instead of `-` -- you'll need to deal with this somehow when generating the `option_#{option_name}` functions. Maybe name `option_plus_#{option_name}` instead for those?

Reply:
Also I'd expect that functions with hyphens in the middle of the operator, such as `gaussian-blur`, would be converted to underscores: `gaussian_blur`, so probably deal with that here too.

There is [String.replace_leading/3](https://hexdocs.pm/elixir/String.html#replace_leading/3) to help replace the leading `-` or `+` independently from the hyphens in the middle of the operator.